src/integrationTest/cloudformation/templateRegistryManager.test.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as path from 'path'
+import * as vscode from 'vscode'
+
+import { CloudFormationTemplateRegistry } from '../../shared/cloudformation/templateRegistry'
+import { CloudFormationTemplateRegistryManager } from '../../shared/cloudformation/templateRegistryManager'
+import { rmrf } from '../../shared/filesystem'
+import { getLogger, setLogger } from '../../shared/logger/logger'
+import { makeSampleSamTemplateYaml, strToYamlFile } from '../../test/shared/cloudformation/cloudformationTestUtils'
+import { TestLogger } from '../../test/testLogger'
+import { getTestWorkspaceFolder } from '../integrationTestsUtilities'
+
+/**
+ * Note: these tests are pretty shallow right now. They do not test the following:
+ * * Adding/removing workspace folders
+ */
+describe.only('CloudFormation Template Registry Manager', async () => {
+    let registry: CloudFormationTemplateRegistry
+    let manager: CloudFormationTemplateRegistryManager
+    let workspaceDir: string
+    let testDir: string
+    const additionalFiles: string[] = []
+
+    before(async function() {
+        try {
+            getLogger()
+        } catch (e) {
+            setLogger(new TestLogger())
+        }
+        workspaceDir = getTestWorkspaceFolder()
+        testDir = path.join(workspaceDir, 'cloudFormationTemplateRegistry')
```

Comment:
Can/Should testDir be deleted after all tests?

Reply:
Potentially? As of now, it's baked into the codebase.

Reply:
Oh right, I forgot about that. Is it possible for these tests to write to a different directory instead of cloudFormationTemplateRegistry? I believe the sam init integ tests write to a separate folder

Reply:
I might be able to create a temp folder and write the files there. The whole key is that the files need to exist in the dir before the watcher is applied for this test. The test files here are the same YAML text as what I'm writing with commands elsewhere here.

Reply:
Consider giving each test run a unique testDir